backend/flashfolio/deck.go

```diff
@@ -157,6 +156,59 @@ func CloneDeckReq(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(ret)
 }
 
+func DeleteDeckReq(w http.ResponseWriter, r *http.Request) {
+
+	reqBody, err := ioutil.ReadAll(r.Body)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var req struct {
+		Deck  Deck   `json:"Deck"`
+		Token string `json:"Token"`
+	}
+
+	json.Unmarshal(reqBody, &req)
+
+	tokenInfo, err := VerifyIdToken(req.Token)
+	if err != nil {
+		/* Bad Token => Unauthorized */
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	user, err := GetUserByID(tokenInfo.UserId, ctx)
+
+	var savedDeck Deck
+	err = DeckCollection.FindOne(ctx, bson.D{{Key: "id", Value: req.Deck.ID}}).Decode(&savedDeck)
+	if err != nil {
+		/* That deck isn't saved to the database*/
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	DeckCollection.DeleteOne(ctx, bson.D{{Key: "id", Value: req.Deck.ID}})
+
+	var index *int
+	for i := 0; i < len(user.OwnedDecks); i++ {
+		if user.OwnedDecks[i] == req.Deck.ID {
+			foundIndex := i
+			index = &foundIndex
+		}
+	}
+	if index != nil {
+		user.OwnedDecks[*index] = user.OwnedDecks[len(user.OwnedDecks)-1]
+		user.OwnedDecks = user.OwnedDecks[:len(user.OwnedDecks)-1]
+	}
```

Comment:
These changes to the user struct are never actually written to the database, meaning dead decks will appear on the profile page.
![Screenshot from 2021-12-02 11-57-20](https://user-images.githubusercontent.com/78443917/144471025-0ce6d5b3-af62-4f75-bb13-7b52ba978cc6.png)

Reply:
Fixed.